src/System.Private.CoreLib/shared/System/Enum.cs

```diff
@@ -906,6 +906,57 @@ public override string ToString()
             // Try to see if its one of the enum values, then we return a String back else the value
             return InternalFormat((RuntimeType)GetType(), ToUInt64()) ?? ValueToString();
         }
+
+        public int CompareTo(object? target)
+        {
+            if (target == this)
+                return 0;
+
+            if (target == null)
+                return 1; // all values are greater than null
+
+            if (this.GetType() != target.GetType())
+                throw new ArgumentException(SR.Format(SR.Arg_EnumAndObjectMustBeSameType, target.GetType(), this.GetType()));
+
+            ref byte pThisValue = ref this.GetRawData();
+            ref byte pTargetValue = ref target.GetRawData();
+
+            switch (this.InternalGetCorElementType())
+            {
+                case CorElementType.ELEMENT_TYPE_I1:
+                    return Unsafe.As<byte, sbyte>(ref pThisValue).CompareTo(Unsafe.As<byte, sbyte>(ref pTargetValue));
```

Comment:
This has a minor observable change - the return value is not normalized to 1/0/-1. This should be acceptable. We have done similar changes for other CompareTo methods in .NET Core.